src/main/java/teammates/ui/template/CourseEditSectionRow.java

```diff
@@ -144,9 +144,15 @@ public ElementTag getToggleSessionLevelInSectionButton() {
             for (int j = 0; j < 3 && i + j < sectionNames.size(); j++) {
                 String name = Const.ParamsNames.INSTRUCTOR_SECTION_GROUP + sectionIndex
                               + Const.ParamsNames.INSTRUCTOR_SECTION + (i + j);
-                ElementTag checkbox = createCheckBox(sectionNames.get(i + j), name, sectionNames.get(i + j),
-                                                     i + j == sectionIndex);
-                specialSectionGroup.add(checkbox);
+                if (isSectionSpecial()) {
+                    ElementTag checkbox = createCheckBox(sectionNames.get(i + j), name, sectionNames.get(i + j),
+                                                         i + j == sectionIndex);
```

Comment:
Maybe extract out <code>i + j == sectionIndex</code> to an explaining variable? The current code is very hard to understand, let's try to improve it if we can.